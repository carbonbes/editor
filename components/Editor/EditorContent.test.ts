import { describe, expect, test } from 'vitest'
import { flushPromises, mount } from '@vue/test-utils'
import EditorContent from '~/components/Editor/EditorContent.client.vue'
import type { Content } from '@tiptap/vue-3'

describe('EditorContent', () => {
  test('should be defined', () => {
    expect(EditorContent).toBeDefined()
  })

  describe('renders the correct markup', () => {
    async function mountEditorContent(content?: Content) {
      const wrapper = mount(EditorContent, { props: { content } })

      await flushPromises()

      return wrapper
    }

    const cases = [
      ['without content'],
      ['with content', '<h1>Всем привет!</h1>'],
    ]

    test.each(cases)('%s', async (_, content) => {
      const wrapper = await mountEditorContent(content)

      expect(wrapper.element.innerHTML).toMatchSnapshot()

      wrapper.unmount()
    })
  })
})
