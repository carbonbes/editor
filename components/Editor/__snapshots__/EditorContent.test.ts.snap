// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`EditorContent > renders the correct markup > with content 1`] = `"<div contenteditable="true" translate="no" class="tiptap ProseMirror h-full overflow-x-hidden prose touch-pan-y focus:outline-none [&amp;>*]:relative [&amp;>.ProseMirror-selectednode]:after:absolute [&amp;>.ProseMirror-selectednode]:after:inset-0 [&amp;>.ProseMirror-selectednode]:after:-m-2 [&amp;>.ProseMirror-selectednode]:after:bg-blue-50 [&amp;>.ProseMirror-selectednode]:after:rounded-xl [&amp;>.ProseMirror-selectednode]:after:z-[-1] [&amp;>*]:transition-transform" tabindex="0"><h1>Всем привет!</h1></div>"`;

exports[`EditorContent > renders the correct markup > without content 1`] = `"<div contenteditable="true" translate="no" class="tiptap ProseMirror h-full overflow-x-hidden prose touch-pan-y focus:outline-none [&amp;>*]:relative [&amp;>.ProseMirror-selectednode]:after:absolute [&amp;>.ProseMirror-selectednode]:after:inset-0 [&amp;>.ProseMirror-selectednode]:after:-m-2 [&amp;>.ProseMirror-selectednode]:after:bg-blue-50 [&amp;>.ProseMirror-selectednode]:after:rounded-xl [&amp;>.ProseMirror-selectednode]:after:z-[-1] [&amp;>*]:transition-transform" tabindex="0"><p><br class="ProseMirror-trailingBreak"></p></div>"`;
